rfcs/modal-component.md

```diff
@@ -20,29 +20,23 @@
 
 ### <Modal/> API
 
-- `children: React.node | ({close: fn}) => React.node` - Required
-  Modal content. The children-as-function API may be preferable for performance reasons (wont render until opened) and also gives users access to a `close()` callback that they can add to buttons
+- `children: React.node | () => React.node` - Required
+  Modal content. The children-as-function API may be preferable for performance reasons (wont render until opened)
 - `isOpen: boolean` - Required
   Controls whether the modal is visible or not
 - `closeable: boolean` - Optional, Defaults to `true`
   Whether the modal should be closeable by the user (either via escape, backdrop click, etc). You can set this to `false` if your modal has an action that the user _must_ take before closing.
 - `onClose: (source: CloseSource) => void` - Optional
   A callback that is invoked when the modal will close. Callback is passed a constant identifying what triggered the close.
-- `size: SIZE.{small|medium|large|full|cover|auto}|string|number`, Defaults to SIZE.default
+- `size: SIZE.{default|full|auto}|string|number`, Defaults to SIZE.default
   Controls the size of the modal (primarily width). Can be a SIZE constant or css `width` property value.
-- `aria-labelledby: string` - Optional, Default to ''
-  If specified, will be passed to the dialog element
-- `aria-describedby: string` - Optional, Defaults to ''
-  If specified, will be passed to the dialog element
-- `aria-label: string` - Optional, Defaults to ''
-  If specified, will be passed to the dialog element (described the modal for screen readers)
-- `id: string` - Optional, Defaults to ''
-  If specified, will be passed to the dialog element
```

Comment:
`aria-labeledby` will be handled in a separate PR. All these others a11y-related props can be passed in via overrides if desired. the modal by default has the necessary default a11y practices without exposing these props explicitly.